src/docs/asciidoc/Optimizing-Serialization.asciidoc

```diff
@@ -0,0 +1,270 @@
+// Empty line above needed after the list from previous file
+[[optimizing-serialization]]
+== Optimizing Serialization
+
+=== Introduction
+
+Spring XD is configured by default to use
+https://github.com/EsotericSoftware/kryo[Kryo] to serialize and deserialize POJO
+message payloads when using remote transport such as Redis, Rabbit, or Kafka.
+Note that if the payload is already in the form of a byte array, then XD will
+transport it as is. Also if the payload is a String, XD will use getBytes() and
+bypass Kryo. Kryo performs favorably compared to other JVM serialization
+libraries but may become a bottleneck in high-throughput applications. In rare
+cases, custom serialization may be needed to address functional issues. This
+section offers some tips and techniques for customizing and optimizing
+serialization to address such situations. In accordance with the usual caveats
+about premature optimization, don't apply these techniques unless you really
+need to. Furthermore, the time it takes to serialize and deserialize an object
+is proportional to factors such the number of fields, composition, and
+collection sizes. Prerequisite to the techniques presented here, consider
+colocating modules (via local binding or module composition), or using lighter
+payloads.
+
+=== Serialization Performance
+
+Published https://github.com/eishay/jvm-serializers/wiki[serialization
+benchmarks] have shown that Kryo serialization operations take less than 1000
+nanoseconds for reasonably sized objects. The Spring XD team has independently
+verified these results. Serializing large objects may take many orders of
+magnitude longer. To put this in perspective, let's suppose that serialization
+and deserialization of your payload each take 5000 nanoseconds.  For a stream
+,e.g., `some-source | some-sink` running in a distributed XD runtime, moving the
+payload from the source to the sink requires serialization at the producer and
+deserialization at the consumer. Both operations would account for 10,000 ns or
+10 microseconds. So the maximum theoretical throughput for this simple stream,
+with no partitioning or parallel deployment, is at most 100,000 messages/sec.
+This does not account for network latency (transporting the serialized bytes),
+overhead in the messaging middleware, or computations performed by the modules.
+If your application requires this level of throughput then serialization is a
```

Comment:
Slight rewording.
If your application requires a higher level of throughput than mentioned above then serialization is a potential bottleneck.